inventory/packages/apt_deb.go

```diff
@@ -104,58 +104,72 @@ func AptUpdates() ([]PkgInfo, error) {
 		Conf linux-image-amd64 (4.9+80+deb9u7 Debian:9.9/stable [amd64])
 	*/
 
-	lines := strings.Split(strings.TrimSpace(string(out)), "\n")
+	lines := strings.Split(strings.TrimSpace(string(data)), "\n")
 
 	var pkgs []PkgInfo
 	for _, ln := range lines {
 		pkg := strings.Fields(ln)
-		if pkg[0] != "Inst" {
+		if len(pkg) < 4 || pkg[0] != "Inst" {
 			continue
 		}
-		if len(pkg) == 5 {
+		if strings.HasPrefix(pkg[2], "(") {
 			// We don't want to record new installs.
 			// Inst firmware-linux-free (3.4 Debian:9.9/stable [all])
 			continue
 		}
-		if len(pkg) != 6 {
-			DebugLogger.Printf("%q does not represent an apt update\n", ln)
+		// Make sure this line matches expectations:
+		// Inst google-cloud-sdk [245.0.0-0] (246.0.0-0 cloud-sdk-stretch:cloud-sdk-stretch [all])
+		if !strings.HasPrefix(pkg[2], "[") || !strings.HasPrefix(pkg[3], "(") || !strings.HasSuffix(pkg[len(pkg)-1], ")") {
```

Comment:
This seems fine but you may want to consider using a regex (or regexes) instead for clarity.

Reply:
The point here is to directly reflect the actions performed below, a regex would be less readable and prone to errors as it would need to allow for lots of variations. We really don't care about any fields other than those listed here (and the first one is a bit redundant with the previous check, so we could drop one of those)